test/test_specs.js

```diff
@@ -238,13 +238,29 @@ suite('specs', function () {
         assert.strictEqual(err, null);
         assert.deepEqual(schema, {
           protocol: 'First',
-          messages: {two: {request: [], response: 'int'}},
-          types: [{name: 'Foo', type: 'fixed', size: 1, namespace: ''}]
+          messages: { two: { request: [], response: 'int' } },
+          types: [{ name: 'Foo', type: 'fixed', size: 1 }]
         });
         done();
       });
     });
 
+    test('imported types get namespace only when not equal to enclosing namespace', function (done) {
+      var opts = {
+        importHook: createImportHook({
+          '1.avdl': '@namespace("com.example.test")\nprotocol First { record Foo { string foo; } }',
+          '2.avdl': '@namespace("com.example.different")\nprotocol Different { record Baz { string baz; } }',
+          '3.avdl': '@namespace("com.example.test")\nprotocol Second { import idl "1.avdl";\nimport idl "2.avdl";\nrecord Bar { string bar; com.example.test.Foo foo; com.example.different.Baz baz; } }'
+        })
+      }
+      assembleProtocol('3.avdl', opts, function (err, schema) {
+        assert.strictEqual(err, null);
+        assert.strictEqual(schema.types.find(type => type.name === "Baz").namespace, "com.example.different")
```

Comment:
The lambdas should be turned to vanilla `function`s. `avsc` is compatible with pre-ES6 versions.